.circleci/soltest.sh

```diff
@@ -92,7 +92,7 @@ do
 done
 
 # wait for individual processes to get their exit status
-for pid in ${PIDs[*]}
+for pid in "${PIDs[*]}"
```

Comment:
Without quotes (original)
```
SC2048 (warning): Use "${array[@]}" (with quotes) to prevent whitespace problems.
```
With quotes (current)
```
SC2066 (error): Since you double quoted this, it will not word split, and the loop will only run once.
```
What's the correct approach to this, and is there actually one other than disabling this check? Also, having looked [shellcheck docs](https://www.shellcheck.net/wiki/Ignore), it seems like an error can only be disable on a per-file basis, and not on a per line basis (more fine grained), which I feel is overkill.

Reply:
This must be either `${PIDs[*]}` (as in the original) or `"${PIDs[@]}"`. The latter will correctly handle array items with spaces in them, though, given that these are process IDs, there should be no spaces and the former will work too. In both cases every PID will be a separate item for the loop.

`"${PIDs[*]}"` is incorrect because it will join the array into a single string (with spaces in it) and will be seen as a single item by the loop.

The gotcha here is simply that the `"${x[@]}"` syntax is counter-intuitive Bash magic. I.e. it has quotes around it so it looks as if the result was a single quoted thing but in fact it will evaluate to multiple separate quoted items.

Reply:
Fixed.